core/src/main/scala/quasar/plugin/jdbc/JdbcDiscovery.scala

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2020 Precog Data
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package quasar.plugin.jdbc
+
+import scala.{Boolean, Char, Int, Nil, None, Option, Some}
+import scala.collection.immutable.Set
+import scala.util.{Either, Left, Right}
+
+import java.lang.String
+
+import cats.~>
+import cats.data.{Ior, NonEmptySet}
+import cats.implicits._
+
+import doobie._
+import doobie.enum.{ColumnNullable, JdbcType}
+import doobie.implicits._
+
+import fs2.Stream
+
+/** Provides a means of discovering schemas and tables in a database. */
```

Comment:
Regarding all the schema-related functionality introduced here: do/will we expose access to tables via database schema in Precog? Or is the schema functionality not going to be used in production yet?

Reply:
@alissapajer It is in the path, we expose the database objects as `/<schema>/<table>`

Reply:
Most of this was cribbed from the Postgres datasource and generalized